src/libraries/System.Security.Cryptography.Cose/tests/CoseSign1MessageTests.Sign.cs

```diff
@@ -0,0 +1,418 @@
+        }
+
+        [Theory]
+        [InlineData(ContentTestCase.Empty)]
+        [InlineData(ContentTestCase.Small)]
+        [InlineData(ContentTestCase.Large)]
+        public void SignWithValidContent(ContentTestCase @case)
+        {
+            byte[] content = GetDummyContent(@case);
+
+            Assert.NotNull(CoseSign1Message.Sign(content, ECDsaKeys[ECDsaAlgorithm.ES256], HashAlgorithmName.SHA256));
+            Assert.NotNull(CoseSign1Message.Sign(content, RSAKey, HashAlgorithmName.SHA256));
+            Assert.NotNull(CoseSign1Message.Sign(content, GetHeaderMapWithAlgorithm((int)ECDsaAlgorithm.ES256), GetEmptyHeaderMap(), ECDsaKeys[ECDsaAlgorithm.ES256]));
+        }
+
+        // Key
+        [Fact]
+        public void SignWithNullKey()
+        {
+            byte[] content = GetDummyContent(ContentTestCase.Small);
+            Assert.Throws<ArgumentNullException>(() => CoseSign1Message.Sign(content, (ECDsa)null!, HashAlgorithmName.SHA256));
+            Assert.Throws<ArgumentNullException>(() => CoseSign1Message.Sign(content, (RSA)null!, HashAlgorithmName.SHA256));
+            Assert.Throws<ArgumentNullException>(() => CoseSign1Message.Sign(content, GetHeaderMapWithAlgorithm((int)ECDsaAlgorithm.ES256), GetEmptyHeaderMap(), null!));
+        }
+
+        [Theory]
+        [InlineData((int)ECDsaAlgorithm.ES256, CoseTestHelpers.SHA256)]
+        [InlineData((int)ECDsaAlgorithm.ES384, CoseTestHelpers.SHA384)]
+        [InlineData((int)ECDsaAlgorithm.ES512, CoseTestHelpers.SHA512)]
+        [InlineData((int)RSAAlgorithm.PS256, CoseTestHelpers.SHA256)]
+        [InlineData((int)RSAAlgorithm.PS384, CoseTestHelpers.SHA384)]
+        [InlineData((int)RSAAlgorithm.PS512, CoseTestHelpers.SHA512)]
+        public void SignWithNonPrivateKey(int algorithm, string hashAlgorithm)
+        {
+            byte[] content = GetDummyContent(ContentTestCase.Small);
+            AsymmetricAlgorithm key;
+
+            if (Enum.IsDefined(typeof(ECDsaAlgorithm), algorithm))
+            {
+                ECDsa ecDsaKey = ECDsaKeysWithoutPrivateKey[(ECDsaAlgorithm)algorithm];
+                Assert.ThrowsAny<CryptographicException>(() => CoseSign1Message.Sign(content, ecDsaKey, new HashAlgorithmName(hashAlgorithm)));
+                key = ecDsaKey;
+            }
+            else
+            {
+                Assert.ThrowsAny<CryptographicException>(() => CoseSign1Message.Sign(content, RSAKeyWithoutPrivateKey, new HashAlgorithmName(hashAlgorithm)));
+                key = RSAKeyWithoutPrivateKey;
+            }
+
+            Assert.ThrowsAny<CryptographicException>(() => CoseSign1Message.Sign(content, GetHeaderMapWithAlgorithm(algorithm), GetEmptyHeaderMap(), key));
+        }
+
+        // SignWithUnsupportedKey (ECDA?)
+        [Fact]
+        public void SignWithUnsupportedKey()
+        {
+            AsymmetricAlgorithm key = DSA.Create();
```

Comment:
DSA is not available on iOS or tvOS, and complicated on macOS (keys can be imported, but not generated).

DSA is more interesting than ECDiffieHellman because it's a signing algorithm (vs a key agreement algorithm), so it "should" work, but it's so old and crufty that nothing new supports it.  But the tests get... weird.

So I'd probably ignore it here and use ECDiffieHellman.